File: Leetcode/Leetcode-DSA-course/Hashing/782-jewels-and-stones/jewels-and-stones.java
class Solution {
    public int numJewelsInStones(String jewels, String stones) {
        Set<Character> jewel_set = new HashSet<>();
        for (char ch : jewels.toCharArray())
            jewel_set.add(ch);

        int ans = 0;
        for (char ch : stones.toCharArray()) {
            if (jewel_set.contains(ch))
                ans++;
        }
        return ans;
    }
}

File: Leetcode/Leetcode-DSA-course/Hashing/782-jewels-and-stones/jewels-and-stones.cpp
class Solution {
public:
    int numJewelsInStones(string jewels, string stones) {
        unordered_set<char> jewel_set(jewels.begin(), jewels.end());
        int ans = 0;
        for (char ch : stones) {
            if (jewel_set.find(ch) != jewel_set.end())
                ans++;
        }
        return ans;
    }
};